code/test/tokens.js

```diff
@@ -332,21 +233,100 @@ contract('ONEWallet', (accounts) => {
+  // Expected result the tokens will be recovered
+  it('TO-BASIC-9 RECOVER_SELECTED_TOKENS: must be able to recover selected tokens', async () => {
+    // create wallets and token contracts used througout the tests
+    let { walletInfo: alice, state: aliceState } = await TestUtil.makeWallet({ salt: 'TO-BASIC-9-1', deployer: accounts[0], effectiveTime: getEffectiveTime(), duration: DURATION })
+    let { walletInfo: carol, state: carolState } = await TestUtil.makeWallet({ salt: 'TO-BASIC-9-2', deployer: accounts[0], effectiveTime: getEffectiveTime(), duration: DURATION, backlinks: [alice.wallet.address] })
+
+    // make Tokens
+    const { testerc20 } = await TestUtil.makeTokens({ deployer: accounts[0], makeERC20: true, makeERC721: false, makeERC1155: false })
+
+    // Begin Tests
+    let testTime = Date.now()
+
+    // Fund Alice with 1000 ERC20 tokens
+    await TestUtil.fundTokens({
+      funder: accounts[0],
+      receivers: [alice.wallet.address],
+      tokenTypes: [ONEConstants.TokenType.ERC20],
+      tokenContracts: [testerc20],
+      tokenAmounts: [[1000]]
+    })
+
+    // Before we can recover we need to track testERC20
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+    await executeTokenTransaction(
+      {
+        ...NullOperationParams, // Default all fields to Null values than override
+        walletInfo: alice,
+        operationType: ONEConstants.OperationType.TRACK,
+        tokenType: ONEConstants.TokenType.ERC20,
+        contractAddress: testerc20.address,
+        testTime
+      }
+    )
+    let trackedTokens = await TestUtil.getTrackedTokensParsed(alice.wallet)
+    Logger.debug(`trackedTokens: ${JSON.stringify(trackedTokens)}`)
+
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+    // Recover test tokens takes an array of uint32 which are the indices of the tracked tokens to recover
+    let hexData = ONEUtil.abi.encodeParameters(['uint32[]'], [[0]])
+    let data = ONEUtil.hexStringToBytes(hexData)
+    let { tx, currentState: aliceCurrentState } = await executeTokenTransaction(
+      {
+        ...NullOperationParams, // Default all fields to Null values than override
+        walletInfo: alice,
+        operationType: ONEConstants.OperationType.RECOVER_SELECTED_TOKENS,
+        dest: carol.wallet.address,
+        data,
+        testTime
+      }
+    )
+    const carolCurrentState = TestUtil.getState(carol.wallet)
+
+    // Validate succesful event emitted
+    TestUtil.validateEvent({ tx, expectedEvent: 'TokenRecovered' })
+
+    // check alices and carols balance
+
+    await TestUtil.validateTokenBalances({
+      receivers: [alice.wallet.address, carol.wallet.address],
+      tokenTypes: [ONEConstants.TokenType.ERC20, ONEConstants.TokenType.ERC20],
+      tokenContracts: [testerc20, testerc20],
+      tokenAmounts: [[0], [1000]]
+    })
+
+    aliceState = await TestUtil.syncAndValidateStateMutation({ wallet: alice.wallet, oldState: aliceState, validateNonce: false })
```

Comment:
was this defined?

Reply:
It was when defining the wallet, but will change it to just be `state` as by default we are using state to refer to alice's state

Reply:
Updated in the latest PR, but this test is still failing, I'll leave this open for now.